### ISSUE TITLE
Bug: (no-useless-escape) throw error in replace $1-$9

### ISSUE THREAD

**Issue by linlinyang** [eslint/eslint#17628]
### Environment

Node version: v16.20.2
npm version: 8.19.4
Local ESLint version: 8.22.0
Global ESLint version: 8.22.0
Operating System: window11

### What parser are you using?

Default (Espree)

### What did you do?

```js
<!-- Paste your code here -->
const str = 'foo{"bar}test';
str.replace(/([{}"])/ig, '\$1');
```

### What did you expect to happen?

don't throw error

### What actually happened?

Unnecessary escape character: \$.eslint[no-useless-escape](https://eslint.org/docs/rules/no-useless-escape)

### Link to Minimal Reproducible Example

https://nolink.com

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ota-meshi** [comment]
Hi @linlinyang
What you want to do is replace it with the string "$1"? If so you should write "$$1".

```js
const str = 'foo{"bar}test';
str.replace(/([{}"])/ig, '$$1'); // "foo$1$1bar$1test"
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_string_as_the_replacement

If that's not the case, I think the behavior will be the same if you remove the `\`. I think you need more explanation of what you want to do.

**Comment by linlinyang** [comment]
@ota-meshi 
I want escape chars of replace } to \}

**Comment by ota-meshi** [comment]
You need to escape `\` and write it as `\\`.

```js
const str = 'foo{"bar}test';
str.replace(/([{}"])/ig, '\\$1'); // foo\{\"bar\}test
```
